locators/patient-master.ts

```diff
@@ -15,20 +22,121 @@ export class PatientMaster {
     this.table = this.page.locator(
       "(//*[@class='table-wrapper'])/div/div/div/div/div/div/table"
     );
+    this.heading = this.page.getByText("Patient Master");
+    this.search = this.page.locator("//*[@data-testid='search-filter-input']");
+    this.sear_result = this.page.locator(
+      "//tbody[@class ='ant-table-tbody']/child::tr/td[2]"
+    );
+    // this.search_key = "James";
+    this.filter = this.page.locator("//*[@data-testid='down-arrow']");
+    this.criteria_dropdown = this.page.locator(
+      "//*[@data-testid='criteria-dropdown']"
+    );
+    this.select_dropdown = this.page.locator(
+      "//div[@data-testid='criteria-value-dropdown']"
+    );
   }
 
-  async searchWithProvidedText(text: string) {
-    await this.searchField.fill(text);
+  async clickOnCreateButton() {
+    await this.createButton.click();
   }
-
   async validateCreatedPatient(text: string) {
     await this.searchField.fill(text);
     await this.page.waitForTimeout(3000);
     let tableData = await this.table.innerText();
     expect(tableData).toContain(text);
   }
 
-  async clickOnCreateButton() {
-    await this.createButton.click();
+  async search_enter(search_key) {
+    this.search_key_recieved = search_key;
+    await this.search.fill(this.search_key_recieved);
+    await this.page.waitForTimeout(10000);
```

Comment:
avoid using static timeout

Reply:
Did not get this ?

Reply:
We are adding unconditional wait. Can we make it conditional such as waiting till an element loads or data loads?

Reply:
Have added waits